qiskit_experiments/data_processing/data_action.py

```diff
@@ -94,11 +86,12 @@ def is_trained(self) -> bool:
         """
 
     @abstractmethod
-    def train(self, data: List[Any]):
+    def train(self, full_val_arr: np.ndarray, full_err_arr: Optional[np.ndarray] = None):
```

Comment:
What if I have a node like a discriminator that trains on data with labels? E.g. IQ data? Perhaps the docs require more examples.